test/doc_formatting_test.ts

```diff
@@ -0,0 +1,45 @@
+import fs from 'fs';
+import glob from 'glob';
+import path from 'path';
+
+/**
+ * Test that catches malformed markdown content in the documentation. This is primarily meant to catch instances where
+ * Markdown auto-formatting breaks our content, which uses non-standard markdown extensions.
+ */
+
+const BaseDir = path.join(__dirname, '..');
+const DocsRoot = path.join(BaseDir, 'docs/');
+const MarkdownPattern = path.join(DocsRoot, '**/*.md');
+
+interface MarkdownFile {
+  path: string;
+  content: string;
+}
+
+async function getMarkdownFileContent(): Promise<MarkdownFile[]> {
+  const files = glob.sync(MarkdownPattern, {});
+  const buffers = await Promise.all(files.map(file => fs.promises.readFile(file)));
```

Comment:
Nit, is this actually better than just doing `fs.readFileSync()` and doing it all synchronously?

Reply:
It is 33% faster, but that's only 30ms 😄  Changed it to sync for simplicity.